### PR TITLE
modification in acos,asin,atan and sqrt function defition

### PR DESCRIPTION
**Pull request by NursenShaikhdm**
modification in acos,asin,atan and sqrt function defition for using safe cast at bigquery

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/type/OperandTypes.java [hunk 1]
```diff
@@ -216,6 +216,11 @@ public static SqlOperandTypeChecker variadic(
   public static final SqlSingleOperandTypeChecker BOOLEAN_BOOLEAN =
       family(SqlTypeFamily.BOOLEAN, SqlTypeFamily.BOOLEAN);
 
+  public static final SqlSingleOperandTypeChecker NUMERIC_BOOLEAN_BOOLEAN =
+      family(ImmutableList.of(SqlTypeFamily.NUMERIC, SqlTypeFamily.BOOLEAN, SqlTypeFamily.BOOLEAN),
+          //2nd and 3rd operands are optional
+          i -> i == 1 || i == 2);
```
Comment (shirin-hasan) [inline core/src/main/java/org/apache/calcite/sql/type/OperandTypes.java:222]: Please check is this condition is required here. 
Supported operand types  for the functions are already mentioned as either Numeric or Numeric_boolean_boolean, implying that 2nd and 3rd operands are optional.
Might work without this condition too.

Comment (NursenShaikhdm) [reply core/src/main/java/org/apache/calcite/sql/type/OperandTypes.java]: yes checked with that and its working so removing that changes

**Comment by shirin-hasan** [comment]
Changes look good to me, in accordance with the changes made on mig.

File: core/src/main/java/org/apache/calcite/sql/fun/SqlStdOperatorTable.java [hunk 2]
```diff
@@ -1661,7 +1663,8 @@ public class SqlStdOperatorTable extends ReflectiveSqlOperatorTable {
           SqlKind.OTHER_FUNCTION,
           ReturnTypes.DOUBLE_NULLABLE,
           null,
-          OperandTypes.NUMERIC,
+          OperandTypes.or(OperandTypes.NUMERIC, OperandTypes.NUMERIC_BOOLEAN,
```
Comment (dm-shantanuc) [inline core/src/main/java/org/apache/calcite/sql/fun/SqlStdOperatorTable.java:1666]: Added test cases for optional boolean parameters for these 4 functions.

**Review by dm-shantanuc — APPROVED** [review]

**Comment by dm-shantanuc** [comment]
Changes are good to merge.

**Pull request merged**